docs/benchmark.md

```diff
@@ -0,0 +1,72 @@
+# Benchmark
+
+Go benchmark tests that measure end-to-end performance of a running Ollama server. Run these tests to evaluate model inference performance on your hardware and measure the impact of code changes.
+
+## When to use
+
+Run these benchmarks when:
+- Making changes to the model inference engine
+- Modifying model loading/unloading logic
+- Changing prompt processing or token generation code
+- Implementing a new model architecture
+- Testing performance across different hardware setups
+
+## Prerequisites
+- Ollama server running locally (`127.0.0.1:11434`)
+- benchstat tool: `go install golang.org/x/perf/cmd/benchstat@latest`
+
+## Usage and Examples
+
+**Note**: All commands must be run from the `benchmark` directory of the Ollama project.
+
+Basic syntax:
+```bash
+go test -bench=. -m MODEL_NAME ./...
+```
+
+Required flags:
+- `-m`: Model name to benchmark
+- `-bench=.`: Run all benchmarks
```

Comment:
what are the benchmarks that can be done individually?

Reply:
The benchmarks are going to be changing a lot over the next few iterations here, so Im gonna leave this one broad for now to avoid letting the docs go out of date.